src/NGen.cpp

```diff
@@ -764,7 +769,47 @@ int main(int argc, char* argv[]) {
 
             int delta_time = manager->Simulation_Time_Object->get_output_interval_seconds();
 
-            router->route(number_of_timesteps, delta_time);
+            std::string t_route_config_file_with_path =
+                manager->get_t_route_config_file_with_path();
+            // model for routing
+            models::bmi::Bmi_Py_Adapter py_troute("T-Route", t_route_config_file_with_path, "troute_nwm_bmi.troute_bmi.BmiTroute", true);
+
+            // tell BMI to resize nexus containers
+            int64_t nexus_count = nexus_indexes.size();
+            py_troute.SetValue("land_surface_water_source__volume_flow_rate__count", &nexus_count);
+            py_troute.SetValue("land_surface_water_source__id__count", &nexus_count);
+            // set up nexus id indexes
+            std::vector<int> nexus_df_index(nexus_count);
+            for (const auto& key_value : nexus_indexes) {
+                int id_index = key_value.second;
+
+                // Convert string ID into numbers for T-route index
+                int id_as_int = -1;
+                int num_char_index = key_value.first.find_first_of("0123456789");
+                if (num_char_index != std::string::npos) {
+                    int num_char_end = key_value.first.find_first_not_of("0123456789");
```

Comment:
Given that the IDs are going to be of the form `nex-1234`, I think this logic is erroneous. Unless I'm misunderstanding, `num_char_end` is going to come out as `0`, and then the call to `substr` will have a negative `count` argument.

To the best of my knowledge, rest of the ngen codebase prefers finding the position of `-` (abstracted as the much longer `hy_features::identifiers::separator`), and then taking the substring of everything after that as being numeric. You could check that said substring really is entirely numeric, but I think you're safe to write the logic as if it will be.